### PR TITLE
[vcpkgTools.xml] Add 7zip binary

### PR DESCRIPTION
**Pull request by JavierMatosD**
This just adds the 7zip binaries to vckpgTools in preparation to use 7zip for all extraction on windows.

### CODE REVIEW

File: scripts/vcpkgTools.xml [hunk 1]
```diff
@@ -144,6 +144,19 @@
         <sha512>6b88a761f7cc8b8affc074b24750bcdc65cc3ab18d127c8bfdd1eca953d43e21558eb1137c4b934689990564d7d24cf14e249a773dc1e5ddb7316b10d73682f8</sha512>
         <archiveName>7z2406-extra.7z</archiveName>
```
Comment (Neumann-A) [inline scripts/vcpkgTools.xml:145]: This is your 7za already. Please update this.

**Comment by Neumann-A** [comment]
Question: 
a) Why add extra stuff to vcpkgTools.xml I thought it should die?
b) Why prefer 7zip over `cmake -E tar` ? (yeah I know 7zip can deal with all those nasty self installing binaries and NIST installers etc.)
c) If you don't want to kill vcpkgTools.xml anymore please add an "arch" entry. That has been missing for ages.

**Comment by JavierMatosD** [comment]
> Question: a) Why add extra stuff to vcpkgTools.xml I thought it should die? b) Why prefer 7zip over `cmake -E tar` ? (yeah I know 7zip can deal with all those nasty self installing binaries and NIST installers etc.) 

As of this moment, I am not sure what the plan is with vcpkgTools.xml. What's motivating this change is we would like to stop using the 7zip.msi and avoid using cmake as a decompressor since 7zip is faster. 

> c) If you don't want to kill vcpkgTools.xml anymore please add an "arch" entry. That has been missing for ages.

My understanding is that vcpkgTools.xml is primarily for internal use and I am not aware of any need for an architecture field. Can you elaborate?

**Comment by Neumann-A** [comment]
> decompressor since 7zip is faster.

Do you have a fair benchmark for that (e.g. not comparing x64 binaries against x86)? The reason cmake is used is to make it easily cross-platform. This is also now possible with 7zip since it provides binaries for !windows. 

I also don't see a reason to give full 7zip in the xml 7zr should be more than enough.

>  primarily for internal use 

What do you mean by internal use? If I don't have a valid cmake version or force exact abi version vcpkg tends to use the CMake from the xml. How is this considered as internal?

> I am not aware of any need for an architecture field

If you build on arm you are basically screwed unless you have a correct version installed. Also using the x86 version of cmake compared to the x64 is annoying since it is only using the x86 since that should work on arm64. As such providing an arch field allows to actually provide the correct binaries for a certain host/build platform.

**Review by BillyONeal — APPROVED** [review]
I agree that this file should die but this is borderline OK as it isn't truly introducing more badness we have to clean up later.

**Pull request merged**